test/test-style.js

```diff
@@ -44,6 +48,48 @@ function escapeInvisibles(str) {
   return finalString;
 }
 
+/**
+ * Gets the row and column matching the index in a string.
+ *
+ * @param {string} str
+ * @param {number} index
+ * @return {[number, number] | [null, null]}
+ */
+function indexToPos(str, index) {
+  let line = 1, col = 1;
+  let prevChar = null;
+
+  if (
+    typeof str !== 'string' ||
+    typeof index !== 'number' ||
+    index > str.length
+  ) {
+    return [null, null];
+  }
+
+  for (let i = 0; i < index; i++) {
+    let char = str[i];
+    switch (char) {
+      case '\n':
+        if (prevChar === '\r') break;
+      case '\r':
+        line++;
+        col = 1;
+        break;
+      case '\t':
+        // Use JSON `tab_size` value from `.editorconfig`
+        col += 2;
+        break;
+      default:
+        col++;
+        break;
+    }
+    prevChar = char;
+  }
+
+  return [line, col];
```

Comment:
Doing it here would break things, as the return type is defined as `[number, number] | [null, null]`.

I’ve instead created a wrapper `indexToPos(…)` function that formats this and renamed the old `indexToPos(…)` to `indexToPosRaw(…)` to be able to get the raw numbers without needing to parse them out from a string.